mona_openai/mona_openai.py

```diff
@@ -0,0 +1,243 @@
+import time
+import asyncio
+from copy import deepcopy
+from types import MappingProxyType
+
+from mona_sdk import MonaSingleMessage
+
+from .exceptions import WrongOpenAIClassException
+from .mona_client import get_mona_clients
+from .util.func_util import (
+    add_conditional_sampling,
+    call_non_blocking_sync_or_async,
+)
+from .endpoints.completion import COMPLETION_CLASS_NAME, get_completion_class
+from .util.validation_util import validate_and_get_sampling_ratio
+
+EMPTY_DICT = MappingProxyType({})
+
+MONA_ARGS_PREFIX = "MONA_"
+CONTEXT_ID_ARG_NAME = MONA_ARGS_PREFIX + "context_id"
+EXPORT_TIMESTAMP_ARG_NAME = MONA_ARGS_PREFIX + "export_timestamp"
+ADDITIONAL_DATA_ARG_NAME = MONA_ARGS_PREFIX + "additional_data"
+
+
+def _get_monitored_base_class(openai_class):
+    """
+    Returns a class that wrapps the given api class with that
+    api-specific functionality.
+    """
+    class_name = openai_class.__name__
+
+    # TODO(Itai): This is essetially a "switch" statement. We should try to
```

Comment:
Why not use a hard-coded dict such that the key is the class name and the value is the relevant function to get the class? then, "if class_name not in CLASS_NAME_TO_FUNC raise exception", otherwise "CLASS_NAME_TO_FUNC[class_name](openai_class)"

Reply:
This is a good idea. It's not completely handling the issue I regard here, but it's a more organized way to do it. Changing.